src/activities/bargame/Bargame.qml

```diff
@@ -0,0 +1,341 @@
+/* GCompris - bargame.qml
+ *
+ * Copyright (C) 2016 UTKARSH TIWARI <iamutkarshtiwari@kde.org>
+ *
+ * Authors:
+ *   Yves Combe (GTK+ version)
+ *   UTKARSH TIWARI <iamutkarshtiwari@kde.org > (Qt Quick port)
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 3 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+import QtQuick 2.1
+import GCompris 1.0
+
+import "../../core"
+import "bargame.js" as Activity
+
+ActivityBase {
+    id: activity
+
+    onStart: focus = true
+    onStop: {}
+
+    pageComponent: Item {
+        id: rootWindow
+        signal start
+        signal stop
+        anchors.fill: parent
+
+        Image {
+            id: backgroundImage
+            source: Activity.url + "school_bg" + Activity.level + ".svg"
+            sourceSize.height: rootWindow.height
+            sourceSize.width: rootWindow.width
+            anchors.fill: rootWindow
+        }
+
+        Component.onCompleted: {
+            activity.start.connect(start)
+            activity.stop.connect(stop)
+        }
+
+        // Add here the QML items you need to access in javascript
+        QtObject {
+            id: items
+            property Item main: activity.main
+            property Item boxModel: boxModel
+            property Item rootWindow: rootWindow
+            property alias tuxRect: tuxRect
+            property alias tux: tux
+            property alias blueBalls: blueBalls
+            property alias greenBalls: greenBalls
+            property alias tuxArea: tuxArea
+            property alias boxes: boxes
+            property alias masks: masks
+            property alias numberOfBalls: numberOfBalls
+            property alias backgroundImage: backgroundImage
+            property alias answerBallsPlacement: answerBallsPlacement
+            property alias answerBalls: answerBalls
+            property alias bar: bar
+            property alias bonus: bonus
+        }
+
+        onStart: {
+            Activity.start(items)
+            Activity.calculateWinPlaces();
```

Comment:
it will be done only once at the beginning, is it what you expect or should it be done at every win/lose?